testutils/resourcebuilder/machine/v1beta1/openstack_provider_spec.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"encoding/json"
+
+	machinev1alpha1 "github.com/openshift/api/machine/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// OpenStackProviderSpec creates a new OpenStack machine config builder.
+func OpenStackProviderSpec() OpenStackProviderSpecBuilder {
```

Comment:
Here I'm not sure if I'm supposed to provide an example of a ProviderSpec that has AZs in it.
If I put AZ, the tests in CPMSO fail.